panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system_ros_interface.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROS_INTERFACE_HPP_
+#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROS_INTERFACE_HPP_
+
+#include <memory>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <realtime_tools/realtime_publisher.h>
+
+#include <std_srvs/srv/trigger.hpp>
+
+#include <panther_msgs/msg/driver_state.hpp>
+
+#include <panther_hardware_interfaces/roboteq_data_converters.hpp>
+
+namespace panther_hardware_interfaces
+{
+
+struct CanErrors
+{
+  bool error;
+  bool write_sdo_error;
+  bool read_sdo_error;
+  bool read_pdo_error;
+  bool front_data_timed_out;
+  bool rear_data_timed_out;
+  bool front_can_net_err;
+  bool rear_can_net_err;
+};
+
+/**
+ * @brief Class that takes care of additional ROS interface of panther system, such as publishing
+ * driver state and providing service for clearing errors
+ */
+class PantherSystemRosInterface
```

Comment:
```suggestion
class PantherSystemRosInterface : public rclcpp::Node
```
If this a ros node we could use standard way of defining it

Reply:
I thought about it and I'm not sure - there is additionally executor inside. It probably will be better to not put executor inside of the node and IMO it makes this class more than a node

Reply:
Was thinking of moving executor out of a class to panther_system but as I'm thinking about it now it adds additional "you need to remember to run executor for this to work". On the other hand you need to add executor always when you want to run ROS node. I don't know now xd

Reply:
I would prefer to leave executor in this class, as the idea behind this class was to abstract whole ROS interface (that isn't supported natively by ros2 control). So it isn't only node, and the class name should indicate that it abstracts whole interface.